main.tf

```
@@ -40,6 +40,42 @@ resource "aws_kms_external_key" "this" {
   tags = var.tags
 }
 
+################################################################################
+# Replica Key
+################################################################################
+
+resource "aws_kms_replica_key" "this" {
+  count = var.create && var.create_replica && !var.create_external && !var.create_replica_external ? 1 : 0
```

could you expand on this logic a bit? It seems like we would just need:
```suggestion
  count = var.create && var.create_replica ? 1 : 0
```

---

nvm - I see what you are doing - you are ensuring that only one key is created per module instantiation.